### PR TITLE
media-gfx/graphviz: Fix a collection of errors

### PR DESCRIPTION
**Pull request by mattst88**
There was a collection of errors that lead here:

- In bug #682520 it was reported that IUSE=gdk-pixbuf didn't add any dependencies, so it should be enabled unconditionally. The maintainer enabled gdk-pixbuf unconditionally (but, see later items) in commit 9d4127b9ff8 ("media-gfx/graphviz: enable gdk-pixbuf unconditionally") rather than recognizing that graphviz was missing a dependency on x11-libs/gdk-pixbuf.

- But in that fix, the maintainer replaced the already-wrong $(use_enable gdk-pixbuf) with --enable-gdk-pixbuf, whereas the correct option was --with-gdk-pixbuf.

- The maintainer apparently recognized that it needed a dependency on x11-libs/gdk-pixbuf, so he added it in commit 39994ea59fb ("media-gfx/graphviz: add missing x11-libs/gdk-pixbuf dep") but for some reason added it to BDEPEND instead of DEPEND/RDEPEND. This was pointed out in 682520#c3 and 682520#c4 but not acted on.

- In bug #828552 it was reported that --enable-gdk-pixbuf was an unknown argument to ./configure. As a result, it was removed in commit 066d7617103 ("media-gfx/graphviz: remove unknown configure arguments"), rather than switching to the correct --with-gdk-pixbuf argument.

gdk-pixbuf is used by the libgvplugin_gdk.so plugin. It is used only if both --with-gdk and --with-gdk-pixbuf are used (both are enabled if not specified).

The --with-gdk option requires gdk-2.0.pc, which is provided by x11-libs/gtk+:2. The --with-gdk-pixbuf option requires gdk-pixbuf-2.0.pc, which is provided by x11-libs/gdk-pixbuf:2. Since gdk-pixbuf is linked in only if x11-libs/gtk+:2 is enabled, control its usage with the existing IUSE=gtk flag.

### CODE REVIEW

File: media-gfx/graphviz/graphviz-2.50.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,274 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (thesamesam) [inline media-gfx/graphviz/graphviz-2.50.0.ebuild:4]: Does 8 work yet?

File: media-gfx/graphviz/graphviz-2.50.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+PYTHON_COMPAT=( python3_{8..10} )
+inherit autotools flag-o-matic java-pkg-opt-2 python-single-r1 qmake-utils
+
+DESCRIPTION="Open Source Graph Visualization Software"
+HOMEPAGE="https://www.graphviz.org/ https://gitlab.com/graphviz/graphviz/"
+# unfortunately upstream uses an "artifact" store for the pre-generated
+# tarball now, which makes predictable URLs impossible.
+SRC_URI="https://gitlab.com/api/v4/projects/4207231/packages/generic/graphviz-releases/${PV}/${P}.tar.xz"
```
Comment (thesamesam) [inline media-gfx/graphviz/graphviz-2.50.0.ebuild:13]: @SoapGentoo we should really ask if upstream would mind doing normal releases

File: media-libs/ganv/ganv-1.8.0-r1.ebuild [hunk 3]
```diff
@@ -1,8 +1,8 @@
-# Copyright 1999-2021 Gentoo Authors
+# Copyright 1999-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
```
Comment (thesamesam) [inline media-libs/ganv/ganv-1.8.0-r1.ebuild:4]: 8?

File: media-gfx/graphviz/graphviz-2.50.0.ebuild (hunk 2 above)
Comment (SoapGentoo) [reply media-gfx/graphviz/graphviz-2.50.0.ebuild:13]: https://gitlab.com/graphviz/graphviz/-/merge_requests/2520#note_881272686

**Review by mikhailkoliada — APPROVED** [review]

File: media-gfx/graphviz/graphviz-2.50.0.ebuild (hunk 1 above)
Comment (mattst88) [reply media-gfx/graphviz/graphviz-2.50.0.ebuild:4]: Yep, changed.

File: media-libs/ganv/ganv-1.8.0-r1.ebuild (hunk 3 above)
Comment (mattst88) [reply media-libs/ganv/ganv-1.8.0-r1.ebuild:4]: Not supported by `waf-utils.eclass`.

**Pull request closed without merging**